src/lib/Log.ml

```diff
@@ -2,7 +2,7 @@ open Basis
 
 type level = [`Info | `Error | `Warn]
 
-type location = LexingUtil.span option
+type location = DriverError.error_with_maybe_span option
```

Comment:
This interface is kind of strange, because `pp_message` is now taking an error, but `pp_message` isn't always about errors... Perhaps I am getting hung up on the names of things.

Perhaps it would be helpful if you could think a bit about how you want all this stuff to work, and then summarize the interface you favor?

Reply:
Log only ever outputs from Driver.process_file... that only ever outputs errors.